Update lambda slot docs to mention lazy rendering (#1552)
ViewComponent/view_component: docs/guide/slots.md

end
```

> Note that while the lambda is called when the `with_*` method is called, if an object that responds to `render_in` is returned (like a component) then it won't be fully rendered until it is first used.

Reviewer: ```suggestion
_Note: While a lambda is called when the `with_*` method is called, a returned component is not rendered until first use._
```
